### PR TITLE
[CARBONDATA-1009] Added check for null value in expression result

### PR DESCRIPTION
**Pull request by kunal642**
Analysis: If the left expression has null value and the right expression has empty string then the equals method would throw NullPointerException when it tries to compare the 2 expressions.

Solution: Add check for null before comparing the objects

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1858/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1859/

**Comment by ravipesala** [comment]
@sounakr Please review this PR

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/detailquery/ExpressionWithNullTestCase.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.apache.carbondata.spark.testsuite.detailquery
```
Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/detailquery/ExpressionWithNullTestCase.scala]: Why empty class is added?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1880/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**